web_generator/README.md

```diff
@@ -1,21 +1,133 @@
+| `include` | Declarations to include in the generated output (as a list). Can either be passed as a raw string to match the full name, or as a regular expression. By defaults it outputs **all exported declarations** | <pre>`include: `<br>` - myNumber`</pre> |
+| `language_version` | The Dart Language Version to use, usually for formatting (optional) | <pre>`language_version: 3.6.0`</pre> |
+| `ts_config` | An object consisting of TS Configurations following the [tsconfig.json file schema](https://www.typescriptlang.org/tsconfig/) used for configuring the TypeScript Program/Compiler (optional) | <pre>`ts_config: `<br>`  compilerOptions: `<br>`    target: es2020`</pre> |
+| `ts_config_file` | The TS Configuration file (tsconfig.json) if any (optional) | <pre>`ts_config_file: tsconfig.json`</pre> |
+| `generate_all` | Include generating declarations for code that isn't exported. Defaults to false | <pre>`generate_all: true`</pre> |
+| `ignore_errors` | Ignore source code warnings and errors (they will still be printed). Defaults to false | <pre>`ignore_errors: true`</pre> |
+| `functions.varargs` | The number of arguments that variable-argument functions should take. Defaults to 4 | <pre>`functions: `<br>`  varargs: 6`</pre> |
+
+### Conventions
+
+The generator scripts use a number of conventions to consistently handle TS
+definitions:
+
+#### Top Level Declarations
+- Top level declarations are handled as top level dart `external` declarations annotated with `@JS`
+
+#### Enums
+- Enums are represented as extension types with static members
+- For most cases, enums are represented without any external calls, and can have their rep type as Dart primitive types where possible
+- If the value for an enum is not given, then the whole enum is treated as an external interop member, and any members without a value are marked as `external`
+
+#### Interfaces and Classes
+- Interfaces and Classes are emitted as extension types that wrap and implement `JSObject`
+- Interface and Class inheritance is maintained using `implements` between extension types
+- Classes have default constructors if none are provided
+- Interface and Classes support members such as: properties, functions, operators, `call` declarations, construct signatures, getters and setters.
+- Readonly properties are represented as getters.
+- Overriding signatures are annotated with [`@redeclare`](https://pub.dev/documentation/meta/latest/meta/redeclare-constant.html). 
+- Multiple instances of interfaces in a given scope are merged together into a single interface. Also supports merging with `var` declarations.
+
+#### Namespaces
+- Namespaces in TS are converted to extension types on `JSObject` with static members.
+- Typed declarations (such as enums, classes, interfaces and namespaces) are prefixed and generated in the general module space.
+- Classes are generated as well as static methods that redirect to constructor calls.
```

Comment:
You already mention that classes are generated in the previous bullet. Prefer: "Constructor calls for classes are generated as static methods."